indexer/services/comlink/src/controllers/api/v4/addresses-controller.ts

```diff
@@ -294,6 +306,72 @@ class AddressesController extends Controller {
       childSubaccounts: subaccountResponses,
     };
   }
+
+  @Post('/:address/registerToken')
+  public async registerToken(
+    @Path() address: string,
+      @Body() body: { token: string, language: string },
+  ): Promise<void> {
+    const { token, language } = body;
+    const foundAddress = await WalletTable.findById(address);
+    if (!foundAddress) {
+      throw new NotFoundError(`No address found with address: ${address}`);
+    }
+    try {
+      const allTokens = await FirebaseNotificationTokenTable.findAll(
+        { address },
+        [],
+      );
+      // Delete tokens older than a month
+      const oneMonthAgo = new Date();
+      oneMonthAgo.setMonth(oneMonthAgo.getMonth() - 1);
+
+      const tokensToDelete = allTokens.filter(
+        (tokenRecord) => new Date(tokenRecord.updatedAt) < oneMonthAgo,
+      );
+
+      if (tokensToDelete.length > 0) {
+        await FirebaseNotificationTokenTable.deleteMany(
+          tokensToDelete.map((tokenRecord) => tokenRecord.token),
+        );
+      }
+      // Register the new token
+      await FirebaseNotificationTokenTable.registerToken(
+        token,
+        address,
+        language,
+      );
+    } catch (error) {
+      throw new DatabaseError(`Error registering token: ${error}`);
+    }
+  }
+
+  @Post('/:address/testNotification')
+  public async testNotification(
+    @Path() address: string,
+  ): Promise<void> {
+    try {
+      const wallet = await WalletTable.findById(address);
+      if (!wallet) {
+        throw new NotFoundError(`No wallet found for address: ${address}`);
```

Comment:
this isn't getting differentiated from other errors, since it's nested in the try clause. Log an error if you want to differentiate it?